lib/utils/src/graph/digraph.cc

```diff
@@ -14,6 +15,32 @@ DirectedEdgeQuery::DirectedEdgeQuery(
     tl::optional<std::unordered_set<Node>> const &dsts)
     : srcs(srcs), dsts(dsts) {}
 
+DirectedEdgeQuery query_intersection(DirectedEdgeQuery const &lhs,
+                                     DirectedEdgeQuery const &rhs) {
+  assert(lhs.srcs.has_value() && lhs.dsts.has_value() && rhs.srcs.has_value() &&
```

Comment:
This assertion should be removed: see https://github.com/flexflow/FlexFlow/blob/6ef694251aea7b4c655f95cdbd4fb1e39e4bbcc5/lib/utils/src/graph/undirected.cc#L14-L24 or the README